src/spi.rs

```diff
@@ -0,0 +1,308 @@
+//! Serial Peripheral Interface (SPI) bus
+
+use crate::stm32::{SPI1, SPI2};
+
+use crate::gpio::gpioa::{PA1, PA11, PA12, PA14, PA15, PA4, PA5, PA6, PA7, PA9};
+use crate::gpio::gpiob::{PB10, PB12, PB13, PB14, PB15, PB2, PB3, PB4, PB5, PB9};
+use crate::gpio::gpioc::{PC1, PC2, PC3};
+use crate::gpio::gpiod::{PD0, PD1, PD3, PD4};
+use crate::gpio::{Alternate, Output, PushPull, AF3, AF5};
+use crate::hal;
+// use crate::hal::blocking::spi::{Read, Write, WriteRead};
+use crate::rcc::Rcc;
+use crate::time::Hertz;
+
+use core::ptr;
+
+pub use crate::hal::spi::{Mode, Phase, Polarity, MODE_0, MODE_1, MODE_2, MODE_3};
+
+/// SPI error
+#[derive(Debug)]
+pub enum Error {
+    Busy,
+    FrameError,
+    /// Overrun occurred
+    Overrun,
+    /// Mode fault occurred
+    ModeFault,
+    /// CRC error
+    Crc,
+    #[doc(hidden)]
+    _Extensible,
```

Comment:
Could we use a [`#[non_exhaustive]`](http://doc.rust-lang.org/reference/attributes/type_system.html#the-non_exhaustive-attribute) here instead?